packages/manager/apps/dedicated/client/app/dedicatedCloud/datacenter/backup/backup-offer.class.js

```diff
@@ -0,0 +1,80 @@
+import get from 'lodash/get';
+import { BACKUP_OFFER_PREMIUM } from './backup.constants';
+
+export default class BackupOffer {
+  constructor({
+    backupDays,
+    customReport,
+    dedicatedProxy,
+    encryption,
+    mailAddress,
+    minimumFullBackups,
+    offerName,
+    price,
+    proxyPerHost,
+    replication,
+    retention,
+    scheduleHour,
+    tax,
+  }) {
+    Object.assign(this, {
+      backupDays,
+      customReport,
+      dedicatedProxy,
+      encryption,
+      mailAddress,
+      minimumFullBackups,
+      offerName,
+      price,
+      proxyPerHost,
+      replication,
+      retention,
+      scheduleHour,
+      tax,
+    });
+  }
+
+  isPremium() {
+    return get(this, 'offerName', null) === BACKUP_OFFER_PREMIUM;
+  }
+
+  hasCustomReport() {
+    return get(this, 'customReport', false);
```

Comment:
I think there are two issues:
- the function is named `has` but returns the content of the property, so maybe we should name it `getCustomReport` instead
- also, why do this ? These are mostly getters, users could do

```js
const backupOffer = new BackupOffer({ customReport: "this is the custom reports" });
if (has(backupOffer, 'customReport')) {
    console.log(backupOffer.customReport);
}
```

Reply:
You are right @FredericEspiau ... We were not using the destructing in the class's constructor, and hence we had to expose getter methods to explicitly tell the user about the properties available. Now that we are using the object destructing in the constructor, the available properties are clearly visible to the user and these getters are no longer required and have been removed. Thanks